python/mxnet/optimizer.py

```diff
@@ -665,26 +667,46 @@ class AdaGrad(Optimizer):
     eps: float, optional
         Small value to avoid division by 0.
     """
-    def __init__(self, eps=1e-7, **kwargs):
+    def __init__(self, eps=1e-7, stype='default', **kwargs):
         super(AdaGrad, self).__init__(**kwargs)
         self.float_stable_eps = eps
+        self.stype = stype
 
     def create_state(self, index, weight):
-        return zeros(weight.shape, weight.context)  # history
+        return zeros(weight.shape, weight.context, stype=self.stype)  # history
```

Comment:
they don't, actually. I get update calls for sized-1 dense weights along with the sparse ones. is the update not expected to occur then? Because the non-sparse version updates them.

Reply:
done